cql3/statements/create_table_statement.cc

```diff
@@ -452,6 +453,20 @@ std::optional<sstring> check_restricted_table_properties(
             break;
         }
     }
+    if (strategy && *strategy == sstables::compaction_strategy_type::time_window) {
+        std::map<sstring, sstring> options = cfprops.get_compaction_type_options();
+        sstables::time_window_compaction_strategy_options _twcs_options(options);
+        auto ttl = cfprops.get_default_time_to_live();
+        int max_windows = qp.db().get_config().twcs_max_window_count();
+
+        if (ttl > 0) {
+            auto window_count = std::chrono::seconds(ttl) / _twcs_options.get_sstable_window_size();
+            if (max_windows > 0 && window_count > max_windows) {
+                throw exceptions::syntax_exception(fmt::format("The number of windows ({}) specified is larger than the allowed "
```

Comment:
In the previous code which I wrote in check_restricted_table_properties() I chose to throw configuration_exception, not syntax_exception. Because it's not really a syntax error, and the query itself might not be an error but it just doesn't fit the configured twcs_max_window_count.